### PR TITLE
Add SetSaslCredentials binding

### PR DESCRIPTION
**Pull request by milindl**
The dotnet binding for rd_kafka_sasl_set_credentials.

This binding is present in the go client only at the moment.

Ref: https://github.com/confluentinc/librdkafka/pull/4033

### CODE REVIEW

File: test/Confluent.Kafka.IntegrationTests/Tests/SetSaslCredentials.cs
Comment (anchitj) [inline test/Confluent.Kafka.IntegrationTests/Tests/SetSaslCredentials.cs:60]: Can we also add overwriting of credentials for Consumer and Admin similar to the Producer above?

**Comment by anchitj** [comment]
Since this is dependent on 2.0.0 librdkafka should we merge this after 2.0.0 is released?

**Comment by milindl** [comment]
Rebased and addressed comments

File: CHANGELOG.md
Comment (anchitj) [inline CHANGELOG.md:1]: [Nit] Better to tag it with next version 2.1.0, we can update it later on if there are some changes in between.

**Review by anchitj — APPROVED** [review]
LGTM. Please update the next version in changelog before merging.

**Pull request merged**
